### DISCUSSION TITLE
Failing to get a clang-sanitize build

### DISCUSSION THREAD

**Discussion by udif** [MikePopoloski/slang#992] [Q&A] [answered]
I'm trying to debug an issue using `clang-sanitize` the same way it is done on CI builds.
I installed `llvm-18` using `wget https://apt.llvm.org/llvm.sh` .
I'm configuring the build using:

`cmake --preset clang-sanitize -DSLANG_CI_BUILD=ON` (I also tried without `-DSLANG_CI_BUILD=ON`)
I've compared the build log output and other than some minor differences (cmake 3.21 vs 3.29, slang6.0.46 vs 6.0.50, different home directories and some spurious whitespaces at the end of some lines), the outputs are identical.

I am building using:
`cmake --build build/clang-sanitize -j8`

At some point I'm getting the following build error:
```
[ 40%] Built target Catch2
[ 41%] Building CXX object source/CMakeFiles/slang_slang.dir/driver/Driver.cpp.o
[ 41%] Building CXX object source/CMakeFiles/slang_slang.dir/driver/SourceLoader.cpp.o
In file included from /home/udif/git/slang/source/diagnostics/Diagnostics.cpp:8:
In file included from /home/udif/git/slang/source/../include/slang/diagnostics/Diagnostics.h:17:
In file included from /home/udif/git/slang/source/../include/slang/numeric/ConstantValue.h:16:
In file included from /home/udif/git/slang/source/../include/slang/numeric/SVInt.h:16:
In file included from /home/udif/git/slang/source/../include/slang/util/SmallVector.h:10:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/algorithm:61:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_algo.h:61:
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:263:8: error: 'get_temporary_buffer<slang::Diagnostic>' is deprecated [-Werror,-Wdeprecated-declarations]
  263 |                 std::get_temporary_buffer<value_type>(_M_original_len));
```
From what I can see `clang` is picking up `gcc` include files, which it doesn't like.
What is the proper way for doing this, assuming I already have `gcc` installed?

**Comment by MikePopoloski** [comment, marked as answer]
It's normal for clang to use libstdc++ (gcc's standard library). It seems like this is specifically some bug with clang & libstdc++12: https://github.com/llvm/llvm-project/issues/76515. I'd say install g++-13 (to get libstdc++13) or downgrade to 11 instead.

**Comment by furkanusta** [comment]
CmakePreset has -Wno-deprecated-declarations for clang, can you check the output whether it is being passed correctly?

**Comment by udif** [comment]
Thanks, I solved it by removing `libstdc++-12-dev` and using `libstdc++-11-dev` (I also removed `gcc++-12` and  `gcc-12` but I'm not sure that was really necessary).
